runtime/colors.rs

```diff
@@ -27,6 +27,57 @@ static NO_COLOR: Lazy<bool> =
 
 static IS_TTY: Lazy<bool> = Lazy::new(|| atty::is(atty::Stream::Stdout));
 
+#[derive(Copy, Clone, Debug)]
+pub enum TTYColorLevel {
+  None,
+  Basic,
+  Ansi256,
+  TrueColor,
+}
+
+static SUPPORT_LEVEL: Lazy<TTYColorLevel> = Lazy::new(detect_color_support);
+
+fn detect_color_support() -> TTYColorLevel {
```

Comment:
To not rely on the environment during testing, we can create a seam at `detect_color_support`:

```suggestion
// this won't compile, but just shows the basic idea
static SUPPORT_LEVEL: Lazy<TTYColorLevel> = Lazy::new(|| detect_color_support(std::env::var_os));

fn detect_color_support(var_os: impl Fn(&str) -> Option<&str>) -> TTYColorLevel {
```

Then in the tests we can do stuff like:

```suggestion
let vars = HashMap::from([
  ("COLORTERM", "truecolor")
]);

assert_eq!(detect_color_support(|name| vars.get(name)), TTYColorLevel::TrueColor);
```